examples/miral-shell/shell_main.cpp

```diff
@@ -83,37 +84,48 @@ int main(int argc, char const* argv[])
                 return true;
 
             case KEY_T:
-                external_client_launcher.launch({terminal_cmd});
+                {
+                    std::vector<std::string> command_line{terminal_cmd};
+                    if (terminal_cmd == "gnome-terminal" && boost::filesystem::exists("/usr/bin/gnome-terminal.real"))
+                    {
+                        // gnome-terminal is a horrid wrapper script on Ubuntu that needs frigging
+                        command_line.emplace_back("--disable-factory");
+                        command_line.emplace_back("--app-id");
+                        command_line.emplace_back("com.canonical.miral.Terminal");
+                    }
+
+                    external_client_launcher.launch(command_line);
+                }
                 return false;
 
-            case KEY_X:
-                external_client_launcher.launch_using_x11({"xterm"});
-                return false;
+                case KEY_X:
```

Comment:
Inconsistent indentation

Reply:
Oops! I forgot to import the Mir layout options to CLion and it went wild!